pkg/runtime/optimize/optimizer.go

```diff
@@ -857,8 +857,21 @@ func (o optimizer) optimizeShowTables(ctx context.Context, stmt *rast.ShowTables
 	return ret, nil
 }
 
-func (o optimizer) optimizeShowIndex(_ context.Context, stmt *rast.ShowIndex, args []interface{}) (proto.Plan, error) {
-	ret := &plan.ShowIndexPlan{Stmt: stmt}
+func (o optimizer) optimizeShowIndex(ctx context.Context, stmt *rast.ShowIndex, args []interface{}) (proto.Plan, error) {
+	var ru *rule.Rule
+	if ru = rcontext.Rule(ctx); ru == nil {
+		return nil, errors.WithStack(errNoRuleFound)
+	}
+
+	vt := ru.MustVTable(stmt.TableName.Suffix())
```

Comment:
If table is non-sharding table,   it will panic.  I think it should be checked first.

Reply:
this top have catch error

Reply:
<img width="1359" alt="image" src="https://user-images.githubusercontent.com/9207285/177260297-15164236-e011-49ae-abf1-f6e8d65e44ad.png">
Without  your fix， non-sharding table can work , for example “show index from departments”. But now it cannot